app/controllers/root_controller.rb

```diff
@@ -164,7 +165,7 @@ def block_empty_format
 
   def prepare_publication_and_environment
     publication = publication_with_places(
-      params[:postcode], params[:slug], params[:edition]
+      PostcodeSanitizer.sanitize(params[:postcode]), params[:slug], params[:edition]
     )
```

Comment:
This is a micro-optimization but it may be worth moving the result of calling `PostcodeSanitizer.sanitize` somewhere where we can memoize the result.  In `publication` we end up doing it twice, once in `prepare_publication_and_environment` and then once immediately below when we assign the result to `@postcode`.